src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketGitSCMRevision.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, Nikolas Falco
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.jenkins.plugins.bitbucket;
+
+import java.text.ParseException;
+import java.util.Date;
+
+import javax.annotation.Nonnull;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.cloudbees.jenkins.plugins.bitbucket.JsonParser.BitbucketDateFormat;
+import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketCommit;
+
+import jenkins.plugins.git.AbstractGitSCMSource.SCMRevisionImpl;
+import jenkins.scm.api.SCMHead;
+
+/**
+ * Represents a specific revision of a bitbucket {@link SCMHead}.
+ *
+ * @author Nikolas Falco
+ * @since 2.2.13
+ */
+public class BitbucketGitSCMRevision extends SCMRevisionImpl {
+    private static final long serialVersionUID = 1L;
+
+    private final String message;
+    private final String author;
+    private final Date date;
+
+    /**
+     * Construct a Bitbucket revision.
+     *
+     * @param head the {@link SCMHead} that represent this revision
+     * @param commit head
+     */
+    public BitbucketGitSCMRevision(@Nonnull SCMHead head, @Nonnull BitbucketCommit commit) {
+        super(head, commit.getHash());
+        this.message = commit.getMessage();
+        this.author = commit.getAuthor();
+        Date commitDate;
+        try {
+            commitDate = new BitbucketDateFormat().parse(commit.getDate());
```

Comment:
I am interested in your feature of traits for skipping CI builds so I built a private copy of this PR as well as the corresponding one in the scm-trait-commit-skip-plugin. We use Bitbucket 4.5 in our CI and this line causes NPE when running this plugin. I added a null check for commit.getDate() and fixed the problem. 

Would you make this change so the feature also works for older versions of Bitbucket?

Reply:
The only reference I can find about the Server version is 5.2.0 in the README.MD file.

I could protect against NPE here but this does not prevent NPE for who use this field because commit date is assumed be always available or trait logic could not work.

Maybe the field was named different in version 4.5 so could you please report the REST payload for your server version?

REST paths for which provide payload (you can clone [this public repo](https://bitbucket.org/nfalco79/test-repos) so I could compare against payload of test case in this PR) are:

1. `https://<server>:<port>/<server_context>/1.0/projects/<owner>/repos/<repo_name>/branches?start=0&limit=200`
2. `https://<server>:<port>/<server_context>/1.0/projects/<owner>/repos/<repo_name>/commits/046d9a3c1532acf4cf08fe93235c00e4d673c1d2`

Reply:
You have a very good point. We are planning a server upgrade in the next couple of months so I think I will just workaround this when you finalize your changes until we have the latest Bitbucket server running. I know I am pushing the limits of compatibility here using such an old Bitbucket server with the latest Jenkins.

For your reference the payload from the 4.5 server with the above REST queries is this after I fed it to `jq .` so it is human readable:
```
{
  "start": 0,
  "values": [
    {
      "isDefault": false,
      "latestChangeset": "bf0e8b7962c024026ad01ae09d3a11732e26c0d4",
      "latestCommit": "bf0e8b7962c024026ad01ae09d3a11732e26c0d4",
      "type": "BRANCH",
      "displayId": "release/release-1",
      "id": "refs/heads/release/release-1"
    },
    {
      "isDefault": false,
      "latestChangeset": "046d9a3c1532acf4cf08fe93235c00e4d673c1d2",
      "latestCommit": "046d9a3c1532acf4cf08fe93235c00e4d673c1d2",
      "type": "BRANCH",
      "displayId": "feature/BB-2",
      "id": "refs/heads/feature/BB-2"
    },
    {
      "isDefault": false,
      "latestChangeset": "fb522a6f08c7c7df337312e4e65ec1b57710672e",
      "latestCommit": "fb522a6f08c7c7df337312e4e65ec1b57710672e",
      "type": "BRANCH",
      "displayId": "feature/BB-1",
      "id": "refs/heads/feature/BB-1"
    },
    {
      "isDefault": true,
      "latestChangeset": "bf4f4ce8a3a8d5c7dbfe7d609973a81a6c6664cf",
      "latestCommit": "bf4f4ce8a3a8d5c7dbfe7d609973a81a6c6664cf",
      "type": "BRANCH",
      "displayId": "master",
      "id": "refs/heads/master"
    }
  ],
  "isLastPage": true,
  "limit": 25,
  "size": 4
}
```
Raw:
```
{"size":4,"limit":25,"isLastPage":true,"values":[{"id":"refs/heads/release/release-1","displayId":"release/release-1","type":"BRANCH","latestCommit":"bf0e8b7962c024026ad01ae09d3a11732e26c0d4","latestChangeset":"bf0e8b7962c024026ad01ae09d3a11732e26c0d4","isDefault":false},{"id":"refs/heads/feature/BB-2","displayId":"feature/BB-2","type":"BRANCH","latestCommit":"046d9a3c1532acf4cf08fe93235c00e4d673c1d2","latestChangeset":"046d9a3c1532acf4cf08fe93235c00e4d673c1d2","isDefault":false},{"id":"refs/heads/feature/BB-1","displayId":"feature/BB-1","type":"BRANCH","latestCommit":"fb522a6f08c7c7df337312e4e65ec1b57710672e","latestChangeset":"fb522a6f08c7c7df337312e4e65ec1b57710672e","isDefault":false},{"id":"refs/heads/master","displayId":"master","type":"BRANCH","latestCommit":"bf4f4ce8a3a8d5c7dbfe7d609973a81a6c6664cf","latestChangeset":"bf4f4ce8a3a8d5c7dbfe7d609973a81a6c6664cf","isDefault":true}],"start":0}
```
And
```
{
  "parents": [
    {
      "parents": [
        {
          "displayId": "8d0fa145bde",
          "id": "8d0fa145bde5151f1d103ab1c3dc1033e6ec4ac1"
        }
      ],
      "message": "Add sample script hello world",
      "authorTimestamp": 1537538845000,
      "author": {
        "emailAddress": "amuniz@example.com",
        "name": "Antonio Muniz"
      },
      "displayId": "bf4f4ce8a3a",
      "id": "bf4f4ce8a3a8d5c7dbfe7d609973a81a6c6664cf"
    }
  ],
  "message": "Add one message more",
  "authorTimestamp": 1537541363000,
  "author": {
    "emailAddress": "nfalco79@acme.com",
    "name": "Nikolas Falco"
  },
  "displayId": "046d9a3c153",
  "id": "046d9a3c1532acf4cf08fe93235c00e4d673c1d2"
}
```
Raw:
```
{"id":"046d9a3c1532acf4cf08fe93235c00e4d673c1d2","displayId":"046d9a3c153","author":{"name":"Nikolas Falco","emailAddress":"nfalco79@acme.com"},"authorTimestamp":1537541363000,"message":"Add one message more","parents":[{"id":"bf4f4ce8a3a8d5c7dbfe7d609973a81a6c6664cf","displayId":"bf4f4ce8a3a","author":{"name":"Antonio Muniz","emailAddress":"amuniz@example.com"},"authorTimestamp":1537538845000,"message":"Add sample script hello world","parents":[{"id":"8d0fa145bde5151f1d103ab1c3dc1033e6ec4ac1","displayId":"8d0fa145bde"}]}]}
```

Reply:
I had compare and the cause is not due the different server version. So maybe you get a bug, could you post the full stacktrace of the NPE to point me on the right direction?

Reply:
Could I think you have a previous commit where there was used the repo.getUpdateOn that was always null?

Reply:
Sorry for the late response. I can try with the merged master branch now and see what the result is.

Reply:
you can always get the latest HPI on incrementals or from Jenkins.
https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/cloudbees-bitbucket-branch-source/2.2.14-rc462.e5e02fcc939c/